services/provider/server/server.go

```diff
@@ -341,77 +345,314 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 	}
 
-	extR = append(extR, &pb.ExternalResource{
-		Name: monConfigMap,
-		Kind: "ConfigMap",
-		Data: mustMarshal(map[string]string{
-			"data":     configmap.Data["data"], // IP Address of all mon's
-			"maxMonId": "0",
-			"mapping":  "{}",
-		})})
-
 	monIps, err := extractMonitorIps(configmap.Data["data"])
 	if err != nil {
 		return nil, fmt.Errorf("failed to extract monitor IPs from configmap %s: %v", monConfigMap, err)
 	}
 
 	extR = append(extR, &pb.ExternalResource{
 		Kind: "CephConnection",
-		Name: "monitor-endpoints",
+		Name: consumerResource.Status.Client.Name,
 		Data: mustMarshal(&csiopv1a1.CephConnectionSpec{Monitors: monIps}),
 	})
 
-	scMon := &v1.Secret{}
-	// Secret storing cluster mon.admin key, fsid and name
-	err = s.client.Get(ctx, types.NamespacedName{Name: monSecret, Namespace: s.namespace}, scMon)
+	consumerConfigMap := &v1.ConfigMap{}
+	consumerConfigMap.Name = consumerResource.Status.ResourceNameMappingConfigMap.Name
+	consumerConfigMap.Namespace = consumerResource.Namespace
+	err = s.client.Get(ctx, client.ObjectKeyFromObject(consumerConfigMap), consumerConfigMap)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get %s secret. %v", monSecret, err)
+		return nil, fmt.Errorf("failed to get %s configMap. %v", consumerConfigMap.Name, err)
 	}
 
-	fsid := string(scMon.Data["fsid"])
-	if fsid == "" {
-		return nil, fmt.Errorf("secret %s data fsid is empty", monSecret)
+	// ClientProfile
+	storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
+	if err != nil {
+		return nil, err
 	}
+	var kernelMountOptions map[string]string
+	for _, option := range strings.Split(util.GetCephFSKernelMountOptions(storageCluster), ",") {
+		if kernelMountOptions == nil {
+			kernelMountOptions = map[string]string{}
+		}
+		parts := strings.Split(option, "=")
+		kernelMountOptions[parts[0]] = parts[1]
+	}
+	rbdClientProfileName := consumerConfigMap.Data["rbd-client-profile"]
+	cephfsClientProfileName := consumerConfigMap.Data["cephfs-client-profile"]
+	radosNamespace := consumerConfigMap.Data["rados-namespace-name"]
+	svgName := consumerConfigMap.Data["svg-name"]
+	extR = append(extR,
+		&pb.ExternalResource{
+			Kind: "ClientProfile",
+			Name: rbdClientProfileName,
+			Data: mustMarshal(
+				&csiopv1a1.ClientProfileSpec{
+					Rbd: &csiopv1a1.RbdConfigSpec{
+						RadosNamespace: radosNamespace,
+					},
+					CephConnectionRef: corev1.LocalObjectReference{Name: consumerResource.Status.Client.Name},
+				},
+			),
+		},
+		&pb.ExternalResource{
+			Kind: "ClientProfile",
+			Name: cephfsClientProfileName,
+			Data: mustMarshal(
+				&csiopv1a1.ClientProfileSpec{
+					CephFs: &csiopv1a1.CephFsConfigSpec{
+						SubVolumeGroup:     svgName,
+						KernelMountOptions: kernelMountOptions,
+						RadosNamespace:     ptr.To(svgName),
+					},
+					CephConnectionRef: corev1.LocalObjectReference{Name: consumerResource.Status.Client.Name},
+				},
+			),
+		},
```

Comment:
1. This will always produce too different client profiles, which does not align with our design where we will only produce a single client profile if the name are the same. 
2. This code does not address the NFS client profile handling